packages/core/entity/FileExplorerURL/index.ts

```diff
@@ -157,4 +157,79 @@ export default class FileExplorerURL {
             sortColumn,
         };
     }
+
+    public static convertToPython(urlComponents: Partial<FileExplorerURLComponents>) {
+        const collectionString = this.convertCollectionToPython(urlComponents?.collection);
+
+        const groupByQueryString =
+            urlComponents.hierarchy
+                ?.map((annotation) => this.convertGroupByToPython(annotation))
+                .join("") || "";
+
+        // Group filters by name and use OR to concatenate same filter values
+        const filterGroups = new Map();
+        urlComponents.filters?.forEach((filter) => {
+            const pythonQueryString = filterGroups.get(filter.name);
+            if (!pythonQueryString) {
+                filterGroups.set(filter.name, this.convertFilterToPython(filter));
+            } else {
+                filterGroups.set(
+                    filter.name,
+                    pythonQueryString.concat(` | ${this.convertFilterToPython(filter)}`)
+                );
+            }
+        });
+
+        // Chain the filters together
+        let filterQueryString = "";
+        filterGroups.forEach((value) => {
+            filterQueryString = filterQueryString.concat(`.query('${value}')`);
+        });
+
+        const sortQueryString = urlComponents.sortColumn
+            ? this.convertSortToPython(urlComponents.sortColumn)
+            : "";
+        // const fuzzy = [] // TO DO: support fuzzy filtering
+
+        const hasQueryElements = groupByQueryString || filterQueryString || sortQueryString;
+        const imports = "import pandas\n";
+        const comment = hasQueryElements ? "#Query on dataframe df" : "#No options selected";
+        const fullQueryString = `${comment}${
```

Comment:
We may want to set the output of the query to a variable for users:
```
import pandas
#Convert current datasource file to a pandas dataframe
df = pandas.read_csv('C:\Users\Brian\Desktop\Variance Paper Dataset.csv').astype('str')

#Query on dataframe df
df.groupby('Plate Barcode', group_keys=True).apply(lambda x: x)
```
```
import pandas
#Convert current datasource file to a pandas dataframe
df = pandas.read_csv('C:\Users\Brian\Desktop\Variance Paper Dataset.csv').astype('str')

#Query on dataframe df
df_queried= df.groupby('Plate Barcode', group_keys=True).apply(lambda x: x)
```

Reply:
Added in [4c46839](https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/98/commits/4c46839ed84944cd4653718cf96ba1c93cd5d293)